### PR TITLE
feat: add moonpay fiat onramp

### LINKED ISSUE #2068: Add MoonPay to the fiat onramp providers
MoonPay will have STX available in "early Q1". We can start the integration into their widget any time in preparation for it, so we can ship as soon as they are ready

### PR DESCRIPTION
**Pull request by beguene**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/stacks-wallet-web/actions/runs/2035702901).

closes #2068

Before **merging** this into dev, we must release this https://github.com/hirosystems/stacks-wallet-web/pull/2313
We can't pass the walletAddress for moonpay becase they require a signature of the URL that has to be done server-side and we don't have a server...

### CODE REVIEW

**Review by fbwoolf — APPROVED** [review]
Nice!

**Comment by fbwoolf** [comment]
I just realized this is in `Draft` mode, sorry I reviewed.

**Comment by beguene** [comment]
@fbwoolf No this is great, thanks for the review. I placed it in draft so we don't merge it.

**Review by landitus — DISMISSED** [review]
👋 I've noticed some visual bugs, mostly regarding the page layout:

1. The page is missing it's top margin (regression)
2. The page is missing it's bottom margin, noticiable when scrolling down (regression)
3. This was already happening, but just noticed the layout works different here than in other sections. It seems the content is being centered vertically, which doesn't give a good result when reducing the viewport vertically. Note that some other pages have a better experience, such as **Send**.

Screenshot 2022-03-18 at 15 35 39
Screenshot 2022-03-18 at 15 38 23
Screenshot 2022-03-18 at 15 47 34

**Comment by timstackblock** [comment]
I do not see STX listed in the asset list on moonpay

Screen Shot 2022-03-21 at 9 49 39 AM

**Comment by landitus** [comment]
As @fbwoolf has been improving the responsiveness of the app in a separate PR, the first step would be to review this https://github.com/hirosystems/stacks-wallet-web/pull/2277 and check if the visual bugs are fixed there.

**Comment by landitus** [comment]
I can confirm that https://github.com/hirosystems/stacks-wallet-web/pull/2277 fixes the visual bugs I've found above. I've tested by adding more content with Chrome dev tools, and as result, the margins and scrolling work as they should. cc/ @fbwoolf 

https://user-images.githubusercontent.com/239215/159284823-e351550f-d9d6-44d3-8d88-095c8e713e5a.mp4

**Comment by Eshwari007** [comment]
Couple of Test Findings:

1- The asset STX is not to be found in Moonpay as indicated by @timstackblock above
2- When the user selects a USDC coin (or any ETH based token) and  if the user has Metamask installed, the user is directed to Metamask
3- When the user removes Metamask from the ext and selects the USDC coin, the user receives 'Enter wallet address "page view
4. Am aware that we don't have a backward arrow to go back to Hiro wallet when the user selects the fiats on-ramp such as Transak, Moonpay, and Okcoin but believe we have a ticket for it which will ease user interface cc @markmhx 
5. Only on Okcoin integration, we have a Hiro page link header while we don't have it for Transak nor Moonpay, for consistency wise-would think it good to have Hiro page link on Moonpay and Transak as well.

cc @markmhx @beguene @timstackblock @andresgalante 
Attaching screen recordings and screenshots of the findings above.



https://user-images.githubusercontent.com/38872890/159292430-6090c5eb-5681-4cdb-909d-1efe0e2850b4.mov

Screen Shot 2022-03-21 at 11 08 14 AM

**Comment by markmhendrickson** [comment]
There's no need to QA this PR until Moonpay has released support for Stacks.

**Comment by andresgalante** [comment]
@landitus to create the video and send it to moonpay

**Comment by fbwoolf** [comment]
I rebased this on the new home page work and changed the base branch. That PR should be merged prior to this.

**Comment by fbwoolf** [comment]
@landitus once you unblock this, I can merge it into dev and create a new release PR.

**Comment by landitus** [comment]
There's one last change we discussed with @markmhx. We'd like to switch the order of the providers on the **Fund account** page. The new order should be:

1. Moonpay
2. Okcoin
3. Transak

**Comment by fbwoolf** [comment]
> There's one last change we discussed with @markmhx. We'd like to switch the order of the providers on the **Fund account** page. The new order should be:
> 
> 1. Moonpay
> 2. Okcoin
> 3. Transak

Ok, np, I'll update the order. 👍

File: src/app/query/hiro-config/hiro-config.query.ts
Comment (fbwoolf) [inline src/app/query/hiro-config/hiro-config.query.ts]: Don't forget to change this before merging.

**Comment by fbwoolf** [comment]
@landitus I switched the order on these but was looking at the title and seems really big on this page in comparison to other logged in pages (ex. View Secret Key). Then I realized there is an issue to update the design on this page which has it has 2 columns (like the logged out onboarding), is this something we want to update now or are we rethinking this page? https://github.com/hirosystems/stacks-wallet-web/issues/2107

**Comment by markmhendrickson** [comment]
MoonPay is saying we can and should pass the wallet address as a GET parameter (`walletAddress`) even though we can't server-side sign it since they'll whitelist it for release (and as such, we don't need to sign it after all).

@fbwoolf can you add this to the integration even if it breaks the flow temporarily (before they end up whitelisting)?

**Comment by landitus** [comment]
> is this something we want to update now or are we rethinking this page?

I noticed the new designs and I believe we should review them first. I'm thinking of adding this as a new issue for a future enhancement. We are going to rethink the funding experience overall as part of the onboarding.

**Comment by markmhendrickson** [comment]
Let's also replace our linking to MoonPay's staging / sandbox site with their production one for development builds (since we can't test effectively on their staging site anyway for lack of STX listing there).

**Comment by markmhendrickson** [comment]
A nitpik to fix that I just noticed: It's "MoonPay" (capitalized "P") not "Moonpay"

**Comment by markmhendrickson** [comment]
Link works for me now during testing!

image

I had to update our admin settings to enable "STX" and disable all other tokens 🤦

**Pull request merged**
